pkg/database/mongoRepository.go

```diff
@@ -109,10 +112,57 @@ func (m *MongoRepository) GetOverallScore(ctx context.Context, userID int64) (fl
 		numLogs++
 	}
 
-	overallScore := totalScore / float64(numLogs)
+	var overallScore int32
+	if numLogs == 0 {
+		overallScore = 0
+	} else {
+		overallScore = int32(math.Round(totalScore / float64(numLogs)))
+	}
+
+	log.Printf("Total sum of log scores for user (ID = %v\n): %v\n", userID, totalScore)
+	log.Printf("Number of total logs for user (ID = %v): %v\n:", userID, numLogs)
+	log.Printf("Average score for user (ID = %v): %v\n:", userID, overallScore)
 
 	return overallScore, err
 }
 
+func (m *MongoRepository) DeleteMentalHealthLogs(ctx context.Context, userID int64, date *pbcommon.Date, all bool) (uint32, error) {
+
+	var filter bson.M // declaring vbariable
+
+	if all {
+		filter = bson.M{"userid": userID} // filtering by user id and date
+	} else {
+		filter = bson.M{"userid": userID, "logdate": date} // filtering by user id and date
+	}
+
+	res, err := m.fileCollection.DeleteMany(ctx, filter) // deleting all health logs with the given date
+
+	if err != nil {
+		log.Fatal("cannot delete mental health logs for user: ", err)
```

Comment:
Do not use Fatal here!!! The server can very easily recover from not deleting mental health logs!! This should be a log to ErrorF and then return the error to the caller to be handled

Reply:
Understood, thank you!